src/test/java/StringManipulationTest.java

```diff
@@ -19,95 +19,193 @@ public void tearDown() {
         manipulatedstring = null;
     }
 
+    //Many word case
     @Test
     public void testCount1() {
         manipulatedstring.setString("This is my string");
         int length = manipulatedstring.count();
         assertEquals(4, length);
     }
 
+    //Empty string case
     @Test
     public void testCount2() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    //Null string case
     @Test
     public void testCount3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString(null); //Not really needed (because string starts null), but kept for clarity.
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    //One word case
     @Test
     public void testCount4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Supercalifragilisticexpialidocious");
+        int length = manipulatedstring.count();
+        assertEquals(1, length);
+    }
+
+    //Only spaces case
+    @Test
+    public void testCount5() {
+        manipulatedstring.setString("           ");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    //Only tab case
+    @Test
+    public void testCount6() {
+        manipulatedstring.setString("\t\t\t");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    //Only carriage return case
+    @Test
+    public void testCount7() {
+        manipulatedstring.setString("\r");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
     }
 
+    //Only newlines case
+    @Test
+    public void testCount8() {
+        manipulatedstring.setString("\n\n\n");
+        int length = manipulatedstring.count();
+        assertEquals(0, length);
+    }
+
+    //Multiple spaces between words case
+    @Test
+    public void testCount9() {
+        manipulatedstring.setString("Lots          of space        between");
+        int length = manipulatedstring.count();
+        assertEquals(4, length);
+    }
+
+    //Do not maintain spacing case
     @Test
     public void testRemoveNthCharacter1() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I' bttr uts0e 16tsinths trn6 rgh?", manipulatedstring.removeNthCharacter(3, false));
     }
 
+    //Maintain spacing case
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
         assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
     }
 
+    //Throws IndexOutOfBoundsException when n larger than string length
```

Comment:
testing looks good but i would add for exception test  for when it true just to double check if it works